### PR TITLE
4.7 release

### PR DESCRIPTION
**Pull request by terakilobyte**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - N/A
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/4.7-release/whats-new/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: snooty.toml [hunk 1]
```diff
@@ -17,9 +17,9 @@ sharedinclude_root = "https://raw.githubusercontent.com/10gen/docs-shared/main/"
...
 driver-long = "MongoDB Java Driver"
 version = "4.7"
-full-version = "{+version+}.0-beta0"
+full-version = "{+version+}.0"
 
 package-name-org = "mongodb-org"
 api = "https://mongodb.github.io/mongo-java-driver/{+version+}"
 stable-api = "Stable API"
-mongocrypt-version="1.4.0" # ensure compatibilty with driver version
+mongocrypt-version = "1.4.0"                                    # ensure compatibilty with driver version
```
Comment (ccho-mongodb) [inline snooty.toml:25]: Question: is the whitespace between the setting value and the comment necessary?

File: source/whats-new.txt [hunk 2]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
 * :ref:`Version 4.0 <version-4.0>`
 
+.. _version-4.7:
+
+What's New in 4.7
+-----------------
+
+New features of the 4.7 driver release include:
+
+- Improved change stream support
```
Comment (ccho-mongodb) [inline source/whats-new.txt:31]: Issue: style guide "Introduce a list with a sentence, and end the sentence with a colon."
Suggestion: "Added the following change stream support improvements:"

File: source/whats-new.txt [hunk 3]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+
+  - Enhanced pre-image and post-image support for :ref:`change streams <java-usage-watch>`.
+    Change stream watch helpers now accept ``whenAvailable`` and ``required``
+    for the fullDocument option.
+  - Added a new ``fullDocumentBeforeChange`` option, which accepts
+    ``whenAvailable`` and ``required``.
+  - Change events may now include a ``fullDocumentBeforeChange`` response field.
```
Comment (ccho-mongodb) [inline source/whats-new.txt:38]: Suggestion: clarify what it means to "may now include" as it doesn't seem definitive.

File: source/whats-new.txt [hunk 4]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+  - Added ``showExpandedEvents`` support for change streams.
+  - Added ``wallTime`` support in ``ChangeStreamDocument`` class.
```
Comment (ccho-mongodb) [inline source/whats-new.txt:40]:
```suggestion
  - Added ``wallTime`` support in the ``ChangeStreamDocument`` class.
```

File: source/whats-new.txt [hunk 5]
```diff
@@ -20,6 +21,48 @@ Learn what's new in:
...
+.. _version-4.7:
+
+What's New in 4.7
+-----------------
+
+New features of the 4.7 driver release include:
+
+- Improved change stream support
+
+  - Enhanced pre-image and post-image support for :ref:`change streams <java-usage-watch>`.
```
Comment (jyemin) [inline source/whats-new.txt:33]: Can any of these things link to documentation that describes the feature in general (driver-agnostically)?

File: source/whats-new.txt [hunk 6]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+    Change stream watch helpers now accept ``whenAvailable`` and ``required``
+    for the fullDocument option.
+  - Added a new ``fullDocumentBeforeChange`` option, which accepts
+    ``whenAvailable`` and ``required``.
+  - Change events may now include a ``fullDocumentBeforeChange`` response field.
+  - Added ``showExpandedEvents`` support for change streams.
+  - Added ``wallTime`` support in ``ChangeStreamDocument`` class.
+
+- Clustered index creation support
```
Comment (ccho-mongodb) [inline source/whats-new.txt]: Issue: Requires a period at the end for style and consistency.

Comment (ccho-mongodb) [reply source/whats-new.txt]: Suggestion: link to the clustered index section in the Fundamentals > Indexes page

File: source/whats-new.txt [hunk 7]
```diff
@@ -20,6 +21,48 @@ Learn what's new in:
...
+  - Change events may now include a ``fullDocumentBeforeChange`` response field.
+  - Added ``showExpandedEvents`` support for change streams.
+  - Added ``wallTime`` support in ``ChangeStreamDocument`` class.
+
+- :ref:`Clustered index <java-clustered-idexes>` creation support
+- The driver supports new features related to :v6.0:`Queryable Encryption </core/queryable-encryption/>`,
+  including support for automatic encryption (enterprise license required)
+  and manual encryption.
+
+  - A new API for key management
```
Comment (jyemin) [inline source/whats-new.txt:47]: These two things are orthogonal to the new Queryable Encryption support.  Not sure they should be nested like this.

File: source/whats-new.txt [hunk 8]
```diff
@@ -20,6 +21,48 @@ Learn what's new in:
...
+  - Support for the new Queryable Encryption shared library, which removes the need
+    for the ``mongocryptd`` process. Queryable Encryption support requires
+    `org.mongodb:mongodb-crypt <https://mvnrepository.com/artifact/org.mongodb/mongodb-crypt>`__
+    version ``1.5.1.1`` or later.
+
+- The driver includes builder API support for a more aggregation stages,
```
Comment (jyemin) [inline source/whats-new.txt:53]: As above: can these link to driver-agnostic docs that describe eash feature in general?

File: source/whats-new.txt [hunk 9]
```diff
@@ -20,6 +21,48 @@ Learn what's new in:
...
+  including ``$search/$searchMeta`` (Atlas only), ``$densify``, and ``$fill``.
+- The driver includes support for a new annotation, ``BsonExtraElements``,
+  that can be used with POJO encoding and decoding. ``BsonExtraElements``
+  enables decoding objects that may receive new fields in the future,
+  without requiring developers to explicitly map those new fields.
+- The driver includes performance optimizations, including:
+
+  - The driver includes a new lock-free implementations of the server session pool
```
Comment (jyemin) [inline source/whats-new.txt:61]: Nit: It reads weird to repeat "the driver includes" in each sub-bullet, when it already uses that lingo just above

**Review by jyemin — CHANGES_REQUESTED** [review]

File: source/whats-new.txt [hunk 10]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+    Change stream watch helpers now accept ``whenAvailable`` and ``required``
+    for the fullDocument option.
+  - Added a new ``fullDocumentBeforeChange`` option, which accepts
+    ``whenAvailable`` and ``required``.
+  - Change events may now include a ``fullDocumentBeforeChange`` response field.
+  - Added ``showExpandedEvents`` support for change streams.
+  - Added ``wallTime`` support in ``ChangeStreamDocument`` class.
+
+- Clustered index creation support
+- The driver supports several new features related to :v6.0:`Queryable Encryption </core/queryable-encryption/>`,
```
Comment (ccho-mongodb) [inline source/whats-new.txt]: Suggestion: 1) Keep the structure of the bulleted points consistent with the rest of the bulleted items. 2) Driver is already the implied subject. "Support for several new Queryable Encryption features."

Suggestion: v6.0 directive is no longer needed since the Manual v6.0 has been deployed (earlier today)

File: source/whats-new.txt [hunk 11]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+  including support for automatic encryption (enterprise license required)
```
Comment (ccho-mongodb) [inline source/whats-new.txt:44]:
```suggestion
  including support for automatic encryption (MongoDB v6.0 Enterprise or later is required)
```

File: source/whats-new.txt [hunk 12]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+  and manual encryption.
+- A new API for key management
```
Comment (ccho-mongodb) [inline source/whats-new.txt]: Suggestion: specify what type of key this is for

File: source/whats-new.txt [hunk 13]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+- Support for the new Queryable Encryption shared library, which removes the need
+  for the ``mongocryptd`` process. Queryable Encryption support requires
```
Comment (ccho-mongodb) [inline source/whats-new.txt]:
```suggestion
  for the ``mongocryptd`` process. The Queryable Encryption shared library requires
```

File: source/whats-new.txt [hunk 14]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+  `org.mongodb:mongodb-crypt <https://mvnrepository.com/artifact/org.mongodb/mongodb-crypt>`__
+  version ``1.5.1.1`` or later, which is an optional driver dependency.
```
Comment (ccho-mongodb) [inline source/whats-new.txt]:
```suggestion
  the optional dependency, `org.mongodb:mongodb-crypt <https://mvnrepository.com/artifact/org.mongodb/mongodb-crypt>`__
  version ``1.5.1.1`` or later.
```

File: source/whats-new.txt [hunk 15]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+
+- The driver includes builder API support for a more aggregation stages,
+  including ``$search/$searchMeta`` (Atlas only), ``$densify``, and ``$fill``.
```
Comment (ccho-mongodb) [inline source/whats-new.txt:54]: Issue: grammar

```suggestion
- Added builder API methods for additional aggregation stages,
  including ``$search/$searchMeta`` (Atlas only), ``$densify``, and ``$fill``.
```

File: source/whats-new.txt [hunk 16]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+- The driver includes support for a new annotation, ``BsonExtraElements``,
```
Comment (ccho-mongodb) [inline source/whats-new.txt:55]:
```suggestion
- Added support for a new annotation, ``BsonExtraElements``,
```

File: source/whats-new.txt [hunk 17]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+  that can be used with POJO encoding and decoding. ``BsonExtraElements``
+  is added to enable decoding objects that may receive new fields in the future,
```
Comment (ccho-mongodb) [inline source/whats-new.txt]: Issue: already mentioned it was added.

```suggestion
  enables decoding objects that may receive new fields in the future,
```

File: source/whats-new.txt [hunk 18]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+  without needing to explicitly map those new fields.
+- The driver includes several performance optimizations, including:
```
Comment (ccho-mongodb) [inline source/whats-new.txt]:
```suggestion
- The driver includes several performance optimizations including:
```

File: source/whats-new.txt [hunk 19]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+
+  - The driver includes a new lock-free implementations of the server session pool
+    and the buffer pool.
```
Comment (ccho-mongodb) [inline source/whats-new.txt:62]:
```suggestion
  - Lock-free implementations of the server session pool and the buffer pool.
```

File: source/whats-new.txt [hunk 20]
```diff
@@ -20,6 +21,47 @@ Learn what's new in:
...
+  - The driver includes a new implementation of ``DBCursor`` cleanup that uses Java's
+    Cleaner API instead of finalization, available in Java 9 or later.
```
Comment (ccho-mongodb) [inline source/whats-new.txt:64]:
```suggestion
  - A new implementation of ``DBCursor`` cleanup that uses Java's
    Cleaner API instead of finalization, available in Java 9 or later.
```

**Review by ccho-mongodb — CHANGES_REQUESTED** [review]
Looks mostly good. A few issues/suggestions.

File: snooty.toml (hunk 1 above)
Comment (terakilobyte) [reply snooty.toml:25]: I don't believe it is, that's just how my toml formatter saved it.

File: source/whats-new.txt [hunk 21]
```diff
@@ -20,6 +21,51 @@ Learn what's new in:
...
+    for the ``mongocryptd`` process. The queryable encryption shared library requires
+    `org.mongodb:mongodb-crypt <https://mvnrepository.com/artifact/org.mongodb/mongodb-crypt>`__
+    version ``1.5.1.1`` or later.
+
+- Added a new API for encryption key management.
+
+- Added builder API methods for additional aggregation stages,
+  including :pipeline:`$search`/:pipeline:`$searchMeta` (Atlas only),
+  :pipeline:`$densify`, and :pipeline:`$fill`.
+- Added support for a new annotation, ``BsonExtraElements``,
```
Comment (ccho-mongodb) [inline source/whats-new.txt:59]:
```suggestion
- Added support for a new annotation, ``BsonExtraElements``
```

File: source/whats-new.txt [hunk 22]
```diff
@@ -20,6 +21,51 @@ Learn what's new in:
...
+  that can be used with POJO encoding and decoding. ``BsonExtraElements``
+  enables decoding objects that may receive new fields in the future
+  without requiring developers to explicitly map those new fields.
+- The driver includes performance optimizations including:
```
Comment (ccho-mongodb) [inline source/whats-new.txt:63]:
```suggestion
- Performance optimizations including:
```

File: source/whats-new.txt [hunk 23]
```diff
@@ -20,6 +21,51 @@ Learn what's new in:
...
+
+  - Lock-free implementations of the server session pool and the buffer pool.
+  - A new implementation of ``DBCursor`` cleanup that uses Java's
```
Comment (ccho-mongodb) [inline source/whats-new.txt:66]:
```suggestion
  - A new cleanup implementation of ``DBCursor`` instances that uses Java's
```

**Review by ccho-mongodb — APPROVED** [review]
LGTM, included a few suggestions.

**Review by jyemin — CHANGES_REQUESTED** [review]
A few links seem broken.  I mentioned over Slack.

Also, not sure what we've done for other drivers, but maybe we should have a general note on top that the release fully supports MongoDB 6.0.
